### PR TITLE
Cannot specify atomic counter with location.

### PR DESCRIPTION
**Pull request by ZhiqianXia**
glslang don't report the errors: atomic counter can not specified with location.

```
#version 310 es
precision highp float;
layout(location = 3, binding = 0, offset = 0) uniform atomic_uint uni0;
in highp vec4 dEQP_Position;
out float out0;

void main()
{
out0 = 0.0;
gl_Position = dEQP_Position;
}
```

Signed-off-by: ZhiqianXia <xzq0528@outlook.com>

### CODE REVIEW

File: glslang/MachineIndependent/ParseHelper.cpp [hunk 1]
```diff
@@ -5767,6 +5767,8 @@ void TParseContext::layoutObjectCheck(const TSourceLoc& loc, const TSymbol& symb
                     error(loc, "can only specify on a uniform block", "push_constant", "");
                 if (qualifier.isShaderRecord())
                     error(loc, "can only specify on a buffer block", "shaderRecordNV", "");
+		if (qualifier.hasLocation() && type.isAtomic())
+		    error(loc, "cannot specify atomic counter with location","atomic","");
```
Comment (greg-lunarg) [inline glslang/MachineIndependent/ParseHelper.cpp:5771]: Can you please change message to "cannot specify on atomic counter"? Also, should we replace "atomic" token with "atomic_uint"? Finally, the indentation seems a little off here. Can you please fix?

**Review by greg-lunarg — CHANGES_REQUESTED** [review]
Looks fine functionally. Just some cosmetic tweeks.

Has anyone filed issues against the original CTS code? It would be least disruptive to have those fixed before merging this.

File: glslang/MachineIndependent/ParseHelper.cpp (hunk 1 above)
Comment (ZhiqianXia) [reply glslang/MachineIndependent/ParseHelper.cpp:5771]: @greg-lunarg Thank your commits. I think that  Message to " cannot specify on atomic counter " , token to " location " is better .  That would be more clear for user.

**Review by greg-lunarg — APPROVED** [review]
LGTM. Just waiting for CTS changes to merge.

**Comment by ZhiqianXia** [comment]
@greg-lunarg thank you for review.

**Comment by ZhiqianXia** [comment]
@greg-lunarg Hi greg , It seems not merged to the glslang repository. Could you help me on it ?

**Pull request merged**